### PR TITLE
[RW-6194][risk=no] Pupppeteer runtime-status test rework

### PR DESCRIPTION
**Pull request by aweng98**
`runtime-status-update.spec.ts` is a single test suite contains three very long notebook tests. It is prone to failure.
This PR attempts to build runtime status tests more resilient to failure. Reasons being flaky are:
-- `runtime-status-update.spec.ts` is a single test suite contains three very long notebook tests. Timeout error happens often.
-- Page and test synchronization are off in many functions.

Single test suite is separated into three smaller test suites. The longest two suites are put in `/nightly` directory:
-- `tests/nightly/dataproc-to-gce.spec.ts`
-- `tests/nightly/gce-to-dataproc.spec.ts`
-- `tests/runtime/pause-resume.spec.ts`

New `BaseHelpSidebar`abstract class is for extend by `RuntimePanel` class. Common functions are centrally created in `BaseHelpSidebar` abstract class.

Note: new test suites contains some sleep calls (time in milliseconds). Sleeps are necessary in order to keep tests from being flaky.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -323,7 +323,7 @@ commands:
           command: |
             circleci tests glob "tests/**/*.spec.ts" | circleci tests split --split-by=timings > /tmp/tests-to-run
             yarn test:ci $(cat /tmp/tests-to-run)
-          no_output_timeout: 25m
+          no_output_timeout: 30m
```
Comment (aweng98) [inline .circleci/config.yml:326]: runtime-status tests could be very long due to unpredictability of notebook runtime

File: e2e/app/modal/new-notebook-modal.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default class NewNotebookModal extends Modal {
   }
 
   async name(): Promise<Textbox> {
-    return Textbox.findByName(this.page, {name: 'Name:'});
+    return Textbox.findByName(this.page, {name: 'Name:'}, this);
```
Comment (aweng98) [inline e2e/app/modal/new-notebook-modal.ts:40]: `this` parameter ensures xpath expression contains the parent modal xpath.

File: e2e/app/page/authenticated-page.ts [hunk 3]
```diff
@@ -57,5 +59,12 @@ export default abstract class AuthenticatedPage extends BasePage {
     return getPropValue<string>(username, 'innerText');
   }
 
+  async closeHelpSidebarIfOpen(): Promise<void> {
```
Comment (aweng98) [inline e2e/app/page/authenticated-page.ts:62]: Help Tip sidebar opens automatically when user logins first time. An open sidebar could "confuse" function to use the wrong sidebar when the runtime sidebar is the target.
So if it is found to be open, close it first.

**Comment by aweng98** [comment]
unrelated test failures are being addressed in separate PR https://github.com/all-of-us/workbench/pull/4515

File: e2e/app/page/notebook-page.ts [hunk 4]
```diff
@@ -235,6 +237,28 @@ export default class NotebookPage extends AuthenticatedPage {
     await analysisPage.deleteResource(notebookName, ResourceCard.Notebook);
   }
 
+  /**
+   * Delete runtime
+   */
+  async deleteRuntime(): Promise<void> {
```
Comment (als364) [inline e2e/app/page/notebook-page.ts:243]: Why is this duplicated here?

File: e2e/app/text-labels.ts [hunk 5]
```diff
@@ -93,7 +93,7 @@ export enum LinkText {
    SeeCodePreview = 'See Code Preview',
    StayHere = 'Stay Here',
    Submit = 'Submit',
-   Update = 'Update',
+   Update = 'APPLY & RECREATE',
```
Comment (als364) [inline e2e/app/text-labels.ts:96]: Will this break other tests?

File: e2e/app/component/base-help-sidebar.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import Container from 'app/container';
+import {LinkText, SideBarLink} from 'app/text-labels';
+import {getPropValue} from 'utils/element-utils';
+import Button from 'app/element/button';
+
+const enum Selectors {
+   rootXpath = '//*[@id="help-sidebar"]',
+   // "margin-right: 0px;" is used to deterimine visibility
+   contentXpath = '//*[@data-test-id="sidebar-content" and contains(normalize-space(@style), "margin-right: 0px;")]',
+   deleteXpath = '//*[@role="button"][./*[@alt="Close"]]',
```
[the remark this answers is not in the document]
Comment (aweng98) [reply e2e/app/component/base-help-sidebar.ts:12]: How about `closeIconXpath`?

File: e2e/app/page/notebook-page.ts (hunk 4 above)
Comment (aweng98) [reply e2e/app/page/notebook-page.ts:243]: This runtime sidebar is accessible from the Jupyter Notebook page. Behavior is different when the NotebookPreviewPage is opened automatically. Line #252 and #253 are making sure page synchronized.

File: e2e/app/text-labels.ts (hunk 5 above)
Comment (aweng98) [reply e2e/app/text-labels.ts:96]: Good catch. I created new `ApplyRecreate` enum value, keep `Update` unchanged.

**Review by als364 — APPROVED** [review]
LGTM.

**Pull request merged**